pkg/operator/controller/route-metrics/controller.go

```diff
@@ -0,0 +1,234 @@
+		// Delete the Shard label corresponding to the Ingress Controller from the RouteMetricsControllerRoutesPerShard metric.
+		DeleteRouteMetricsControllerRoutesPerShardMetric(request.Name)
+		log.Info("RoutesPerShard metric label corresponding to the Ingress Controller is successfully deleted", "ingresscontroller", ingressController)
+		return reconcile.Result{}, nil
+	}
+
+	// NOTE: Even though the route admitted status should reflect validity of the namespace and route labelselectors, we still will validate
+	// the namespace and route labels as there are still edge scenarios where the route status may be inaccurate.
+
+	// Get the Namespace LabelSelector from the Ingress Controller.
+	var namespaceLabelSelector map[string]string
+	if ingressController.Spec.NamespaceSelector != nil {
+		namespaceLabelSelector = ingressController.Spec.NamespaceSelector.MatchLabels
+	}
+	// List all the Namespaces which match the Namespace LabelSelector.
+	namespaceList := corev1.NamespaceList{}
+	if err := r.cache.List(ctx, &namespaceList, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(namespaceLabelSelector),
+	}); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to list Namespaces %q: %v", request, err)
+	}
+	// Create a set of Namespaces to easily look up Namespaces that matches the Routes assigned to the Ingress Controller.
+	namespacesSet := sets.NewString()
+	for i := range namespaceList.Items {
+		namespacesSet.Insert(namespaceList.Items[i].Name)
+	}
+
+	// Get the Route LabelSelector from the Ingress Controller.
+	var routeLabelSelector map[string]string
+	if ingressController.Spec.RouteSelector != nil {
+		routeLabelSelector = ingressController.Spec.RouteSelector.MatchLabels
+	}
+	// List all the Routes which matches the Route LabelSelector.
+	routeList := routev1.RouteList{}
+	if err := r.cache.List(ctx, &routeList, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(routeLabelSelector),
+	}); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to list Routes for the Shard %q: %v", request, err)
+	}
+
+	// Variable to store the number of routes admitted by the Shard (Ingress Controller).
+	routesAdmitted := 0
+
+	// Iterate through the list Routes.
+	for _, route := range routeList.Items {
+		// Check if the Route's Namespace matches one of the Namespaces in the set namespacesSet and
+		// the Route is admitted by the Ingress Controller.
+		if namespacesSet.Has(route.Namespace) && routeStatusAdmitted(route, ingressController.Name) {
+			// If the Route is admitted then, the routesAdmitted should be incremented by 1 for the Shard.
+			routesAdmitted++
+		}
+	}
+
+	// Set the value of the metric to the number of routesAdmitted for the corresponding Shard (Ingress Controller).
+	SetRouteMetricsControllerRoutesPerShardMetric(request.Name, float64(routesAdmitted))
+
+	return reconcile.Result{}, nil
+}
+
+// routeStatusAdmitted returns true if a given route has been admitted by the Ingress Controller.
```

Comment:
*nit* a little wording change to describe it's checking the status
```suggestion
// routeStatusAdmitted returns true if a given route's status shows admitted by the Ingress Controller.
```